### PR TITLE
feat: disable 'sendall' when user focuses amount

### LINKED ISSUE #682: Send All
Couple issues with the `Send All` switch when sending:
- When click it again/turn it off, the total amount stays. It should reset the amount.
- When the switch is ON and I change the amount, after clicking on the next field, the amount is changed to total amount. When change the amount, it should turn off the switch.

### PR DESCRIPTION
**Pull request by kalgoop**
## Proposed changes

Send all option should be disabled when user puts the cursor on amount field.
I think, this fixed #682 . 
However, I don't agree with @zillionn that amount should be reset when sendAll is disabled.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes

Also, I have changed name of 'onSendAll' function to 'setSendAll', since this function may be used from other places than send all button.

### CODE REVIEW

**Comment by alexbarnsley** [comment]
@kalgoop 	I sort of agree with @zillion. Could you make it so that when disabling send all, it resets the amount if the amount === the value it would be if send all was enabled please. E.g. if i enable send all and then change the value manually, disabling shouldn't reset the value. if i enable send all, don't change the value and then disable send all, it should reset the value. Let me know if that doesn't make sense please.

**Comment by kalgoop** [comment]
What do you mean by reset ?
Reset to 0 or reset to value that existed before enabling 'send all'

**Comment by alexbarnsley** [comment]
I think reset the field to an empty value so it's more obvious that it's been reset. Both empty and 0 give the same error so don't need to worry about that

**Comment by zillionn** [comment]
IMO reset should be the initial value. Example:
- I click on a `Pay with ARK` link to pay somebody `8.50 ARK`
- Wallet will populate all the fields for me
- Enabling `Send all` will set the amount to my total amount
- Disabling `Send all` should reset to the initial value - `8.50`, not to `0`

**Comment by alexbarnsley** [comment]
I'm happy with that @zillionn, I didn't consider the URI's loading in data to the send form.

**Comment by kalgoop** [comment]
ok. will push after some time

**Comment by alexbarnsley** [comment]
I think keep it as the second point. whatever the initial value was, keep that if resetting. If the form was opened with no amount, reset to no amount. If the form started with an amount, reset to that amount. but only reset if the amount didn't change between enabling and disabling send all

**Comment by kalgoop** [comment]
I'm unable to set field to blank value due to validators.

**Comment by kalgoop** [comment]
I'm unable to set field to blank value (or 0) due to validators.

**Comment by alexbarnsley** [comment]
Maybe try changing the validators so you can

**Comment by Zillion** [comment]
You got the wrong Zillion, you want Zillionn I think. Just noticed I have a
number of emails from you guys.

Kind regards
Zillion with only one n

**Comment by alexbarnsley** [comment]
sorry about that @Zillion with one n 😅

**Comment by zillionn** [comment]
@Zillion Give me your GitHub password and we solve this :wink:

**Comment by kalgoop** [comment]
@alexbarnsley setting amount to blank isn't possible due to number type in v-model.
I was trying since hours to fix this, but couldn't. 

Btw, why do you think current implementation isn't better?
I feel current implementation is better.
Why user would prefer blank field after enabling and disabling sendAll ?
I think there will be more users that would want sendAll amount there after enabling and disabling sendAll with initial blank amount.

**Comment by kalgoop** [comment]
@faustbrian don't you think we should take feedback from users if they feel above is intuitive to them ?

**Comment by kalgoop** [comment]
@alexbarnsley don't you think we should take feedback from users if they feel above is intuitive to them ?

**Comment by zillionn** [comment]
@kalgoop Switching on/off should not break your form and should return to the previous value. Current implementation is broken, If you change the amount while the switch is on, it MUST go off, because currently it stays on and you can't change the amount.

Most people will probably never use the switch, but here is an example:
- You click on `ARK Pay` link, popup will open and most of the fields populated including some amount, then you click `Send All` just to see the total amount in your wallet, because there is no such info in that popup, or misclick it or just out of curiosity, and then what, how you get back the amount from the `ARK Pay` link?! You have to close the popup, search for that exact same link and click it again.

**Comment by alexbarnsley** [comment]
@kalgoop 	is the PR as you envisage it should work? If so I will test and decide from there

**Comment by kalgoop** [comment]
I had tested the code while pushing the pr. And it was as I wanted. 
I can't really test until day after tomorrow, if the conflicts caused some bug.

**Comment by kalgoop** [comment]
@alexbarnsley tested it again today, working as intended.
" If the amount is initially empty, sendAll enabled and disabled -> amount remains the sendAll amount only
If the amount has some value initially, sendAllenabled and disabled -> amount resets to initial value "
Also, when you focus the amount textbox, sendall is disabled.

**Comment by alexbarnsley** [comment]
when focusing on the amount field and enabling/disabling send all, you're getting an error

image

this doesn't happen for me on develop.

**Comment by kalgoop** [comment]
@alexbarnsley fixed all errors.
With this PR - 
-When user toggles off the sendAll, amount is set to previous amount (even if previous amount had errors)
-When user focuses amount, sendAll is disabled, and the amount remains the same (as with sendAll)

File: src/renderer/components/Transaction/TransactionForm/TransactionFormTransfer.vue [hunk 1]
```diff
@@ -39,13 +39,14 @@
         :wallet-network="walletNetwork"
         class="flex-1 mr-3"
         @blur="ensureAvailableAmount"
+        @focus="setSendAll(false,false)"
```
Comment (alexbarnsley) [inline src/renderer/components/Transaction/TransactionForm/TransactionFormTransfer.vue]: There should be a space after the comma `(false, false)`

**Comment by alexbarnsley** [comment]
I don't think it should disable Send All on focus. It should be when the value is changed

**Comment by kalgoop** [comment]
@alexbarnsley I dont agree. When user focuses, he'll come to know that sendAll has been disabled.
If sendAll is disabled on change, then even on focus sendAll is enabled and user may not even try changing the field.

Anyways, lint error and conflicts have been resolved.

**Comment by alexbarnsley** [comment]
Well it seems you disagree with everything I say so I'm going to ask others to review this.

**Comment by kalgoop** [comment]
@alexbarnsley well that's rude. I have given an explanation why I disagree.

**Comment by alexbarnsley** [comment]
How is that rude? I gave my opinions to your changes and you've disagreed with them. I can't move forward with this PR with a stalemate between us, so I will get other people's feedback on your changes.

**Comment by kalgoop** [comment]
@alexbarnsley "disagree with everything I say ".
Okay, leave that.

**Comment by j-a-m-l** [comment]
So guys, how we move forward this PR?

The behaviour described by @ItsANameToo it's what I would do.

**Comment by alexbarnsley** [comment]
@j-a-m-l we'll wait 14 days for a response and then close if there isn't one.

**Pull request merged**